### PR TITLE
Added select options for key and tempo in the view

### PR DESCRIPTION
**Pull request by zhirongg**
The actual filtering functionality has not yet been implemented on the backend.
@leethomas you probably have a better idea of what the sql query and ajax should look like, so feel free to help me implement it :)

### CODE REVIEW

File: app/finders/songs_finder.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class SongsFinder
 
-  def self.search(keywords)
+  def self.search(keywords, key, tempo)
```
Comment (patrickf3139) [inline app/finders/songs_finder.rb]: So I think here, after the full text search, you can filter out the songs that don't have the right key or tempo, assuming they are set.

File: app/controllers/songs_controller.rb [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+    @key_opts = [['Any', '']] + Song::VALID_KEYS.map { |k| [k, k] }
+
     respond_to do |format|
       format.json do
         songs = Song.all
         total_songs = songs.count
-        songs = Song.search_by_keywords(params[:sSearch]) if params[:sSearch].present?
-        
+
+        if params[:search][:value].present?
```
Comment (patrickf3139) [inline app/controllers/songs_controller.rb]: Could `params[:search]` ever be null here and cause a null pointer bug?

Comment (leethomas) [inline app/controllers/songs_controller.rb]: No. it's always 

```ruby
{
  search: { value: "whatever you want" }
}
```

File: test/controllers/songs_controller_test.rb [hunk 3]
```diff
@@ -6,21 +6,42 @@ class SongsControllerTest < ActionController::TestCase
...
-      s.delete('relevance')
-      Song.new(s)
-    end
+    songs_data = load_songs
 
     assert_includes(songs_data, songs(:hands_to_the_heaven))
     assert_includes(songs_data, songs(:glorious_day))
   end
 
+  test "index should stack searches" do
```
Comment (patrickf3139) [inline test/controllers/songs_controller_test.rb]: Perhaps a better description for this is "index should stack filters"

File: test/controllers/songs_controller_test.rb [hunk 4]
```diff
@@ -6,21 +6,42 @@ class SongsControllerTest < ActionController::TestCase
...
+      key: "C",
+      format: :json,
+      xhr: :true
+    }
+
+    get :index, http_params
+    songs_data = load_songs
+
+    assert_includes(songs_data, songs(:all_my_hope))
+    assert_includes(songs_data, songs(:forever_reign))
```
Comment (patrickf3139) [inline test/controllers/songs_controller_test.rb]: What I was saying for this test is you should assert that two songs with two of the three filters matching are NOT in the results. You are checking if songs that match all three filters are in it, but a empty search or one with just one of the filters would return all of them anyway, but also some more, so you want to check that almost matching songs aren't in it.

File: test/controllers/songs_controller_test.rb [hunk 5]
```diff
@@ -6,21 +6,42 @@ class SongsControllerTest < ActionController::TestCase
...
+  end
+
   test "should get new" do
     get :new
     assert_response :success
   end
 
+  private
+  def load_songs
```
Comment (patrickf3139) [inline test/controllers/songs_controller_test.rb]: I love this helper!

File: test/fixtures/songs.yml [hunk 6]
```diff
@@ -141,7 +141,7 @@ forever_reign:
     No other name
             G/B
     Jesus, Jesus
-10_000_reasons:
+ten_thousand_reasons:
```
Comment (patrickf3139) [inline test/fixtures/songs.yml]: Good idea!

**Comment by patrickf3139** [comment]
LGTM! Good work :)

**Pull request merged**